.github/workflows/performance.yml

```
@@ -0,0 +1,110 @@
+name: Performance tests
+
+on:
+  workflow_call:
+    inputs:
+      go-version:
+        required: true
+        type: string
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  performance-test-suite-detect-runners:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.detect-runners.outputs.matrix }}
+    env:
+      PERF_TEST_RUNS_ON: ${{ secrets.PERF_TEST_RUNS_ON }}
+      PERF_TEST_RUNS_ON_DEFAULT: |
+        {
+          "targets": [
+            {
+              "name": "github-ubuntu-latest",
+              "runs-on": "ubuntu-latest"
+            }
+          ]
+        }
+    steps:
+      - id: detect-runners
+        run: |
+          RES="$(echo "${PERF_TEST_RUNS_ON:-${PERF_TEST_RUNS_ON_DEFAULT}}" | jq -c '.targets')"
+          echo "Detected targets:"
+          echo "$RES" | jq .
+          echo "::set-output name=matrix::${RES}"
+
+  performance-test-suite:
+    name: Performance Test Suite (${{ matrix.target.name }})
+    needs:
+      - performance-test-suite-detect-runners
+    strategy:
+      matrix:
+        target: ${{ fromJson(needs.performance-test-suite-detect-runners.outputs.matrix) }}
+    runs-on: ${{ matrix.target.runs-on }}
+    env:
+      ARG_DURATION: "${{ startsWith(github.ref, 'refs/tags/v') && '-d 10m' || '' }}"
+    steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ inputs.go-version }}
+      - uses: actions/checkout@v3
+      - run: go build -o perf-test-suite ./test/performance-test-suite/cmd/perf-test/
+      - name: Run performance tests
+        id: performance
+        run: ./perf-test-suite $ARG_DURATION > perf-test-results.json
+      - name: Upload test results
+        uses: actions/upload-artifact@v3
+        with:
+          name: Performance Test Results (${{ matrix.target.name }})
+          path: perf-test-results.json
+          retention-days: 30
+      - name: Create the Mattermost message
+        if: github.event.schedule == '0 0 * * *'
+        run: >
+          echo "{\"text\":\"### Results for scheduled daily run on branch ${{ github.base_ref }}\n
```

This will be running on different runners, maybe we should also add information which runner it is?

```suggestion
          echo "{\"text\":\"### Results for scheduled daily run on branch ${{ github.base_ref }}, runner ${{ matrix.target.name }}\n
```